File: examples/workspaces/booksandrecords-spring-boot/src/main/java/com/ms/booksandrecords/commands/DealCmd.java
package com.ms.booksandrecords.commands;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
@JsonSubTypes({
        @JsonSubTypes.Type(value = OpenDeal.class, name = "openDeal"),
        @JsonSubTypes.Type(value = CloseDeal.class, name = "closeDeal")
})
public abstract class DealCmd {
}

File: examples/AppModel/booksandrecords-spring-boot/src/main/scala/com/ms/booksandrecords/commands/DealCmd.scala
package com.ms.booksandrecords.commands

import com.fasterxml.jackson.annotation.{JsonSubTypes, JsonTypeInfo}

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
@JsonSubTypes(Array(
  new JsonSubTypes.Type(value = classOf[OpenDeal], name = "openDeal"),
    new JsonSubTypes.Type(value = classOf[CloseDeal], name = "closeDeal")
)
)
trait  DealCmd {
}